programs/ziti-edge-tunnel/windows-service.c

```diff
@@ -200,14 +201,43 @@ VOID SvcInit( DWORD dwArgc, LPTSTR *lpszArgv)
         return;
     }
 
-    // Report running status when initialization is complete.
-
-    ReportSvcStatus( SERVICE_RUNNING, NO_ERROR, 0 );
-
     // start tunnel
     CreateThread (NULL, 0, ServiceWorkerThread, lpszArgv, 0, NULL);
 
-    SvcReportEvent(TEXT("Ziti Edge Tunnel Run"), EVENTLOG_INFORMATION_TYPE);
+    // Check whether the service is started
+
+    // Create an event. The control handler function, SvcCtrlHandler,
+    // signals this event when it receives the running control code.
+
+    ghSvcRunningEvent = CreateEvent(
+            NULL,    // default security attributes
+            TRUE,    // manual reset event
+            FALSE,   // not signaled
+            NULL);   // no name
+
+    if ( ghSvcRunningEvent == NULL)
+    {
+        ReportSvcStatus( SERVICE_STOPPED, GetLastError(), 0 );
+        return;
+    }
+
+    // Report running status when initialization is complete.
+
+    while(1)
+    {
+        // If the service receive a running event with in 5 minutes, set the service to running state
+        // otherwise stop the service
+
+        if (WaitForSingleObject(ghSvcRunningEvent, 300000) == WAIT_OBJECT_0) {
```

Comment:
5 minutes seems like a lot more time that we would want to wait?

Reply:
reduced it to 150 seconds